src/modules/EventModule.cpp

```diff
@@ -0,0 +1,792 @@
+		if (args.size() != 1)
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		auto& config = getServerEventData(channel.guildId());
+		if (config.isEventRunning)
+		{
+			client.createMessage(message.channelId(), "**You can't get free rewards anytime!**");
+			return;
+		}
+		auto& configRD = raffleDrawGuildList[channel.guildId()][raffleDrawGetUserIndex(channel.guildId(), message.author().id())];
+		if (!config.claimedReward)
+		{
+			if(message.author().id() == config.luckyUser)
+			{
+				if (config.raffleDrawRewardClaimTimer->isActive())
+				{
+					CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+					auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+					auto& authorCurrency = currencyModule->guildList[channel.guildId()][currencyModule->getUserIndex(channel.guildId(), message.author().id())];
+					QString authorName = UmikoBot::Instance().GetName(channel.guildId(), message.author().id());
+					config.claimedReward = true;
+					Embed embed;
+					embed.setColor(15844367);
+					authorCurrency.setCurrency(authorCurrency.currency() + 500);
+					embed.setTitle("**Raffle Draw Reward goes to " + authorName + "!**");
+					embed.setDescription("Congratulations **" + authorName + "**!\n"
+										"You just got `500` " + currencyConfig.currencySymbol);
+					config.raffleDrawRewardClaimTimer->stop();
+					client.createMessage(message.channelId(), embed);
+					return;
+				}
+			}
+			else if (config.raffleDrawRewardClaimTimer->isActive())
+			{
+				client.createMessage(message.channelId(), "**BRUH, you don't have the lucky ticket!**");
+				return;
+			}
+		}
+		else
+		{
+			client.createMessage(message.channelId(), "**You can't get free rewards anytime!**");
+		}
+	});
+	RegisterCommand(Commands::EVENT_BUY_TICKET, "buyticket", [this](Client& client, const Message& message, const Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+		auto& config = getServerEventData(channel.guildId());
+		if (!config.eventRaffleDrawRunning)
+		{
+			client.createMessage(message.channelId(), "**RAFFLE DRAW** in not running. You can buy tickets when **RAFFLE DRAW** starts!");
+			return;
+		}
+		if (args.size() != 2)
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		QRegExp re("[+]?\\d*\\.?\\d+");
+		if (!re.exactMatch(args.at(1)))
+		{
+			client.createMessage(message.channelId(), "**You can't buy ticket(s) in invalid amounts**");
+			return;
+		}
+		unsigned int ticket = args.at(1).toUInt();
+		snowflake_t authorID = message.author().id();
+		auto& configRD = raffleDrawGuildList[channel.guildId()][raffleDrawGetUserIndex(channel.guildId(), authorID)];
+		if (configRD.m_TicketIds.size() >= config.maxTicketOfUser)
+		{
+			client.createMessage(message.channelId(), "**You have already bought so many tickets, I can't give you more!**");
```

Comment:
I think both of messages for these checks should be the same (the one below is almost similar, but you forgot a space after the period):
```
"**Don't be selfish!**\nLet others also have a go at some tickets. You can only have " + num + " tickets at max!"
```